pkg/config/config_factory.go

```diff
@@ -35,5 +44,29 @@ func GetConfigProvider() service.ConfigProvider {
 		loc = []string{"env:" + envKey}
 	}
 
-	return service.MustNewDefaultConfigProvider(loc, getSetFlag())
+	providers := []config.MapProvider{filemapprovider.New(), envmapprovider.New(), yamlmapprovider.New()}
+
+	ret := make(map[string]config.MapProvider, len(providers))
```

Comment:
What is `ret` supposed to symbolize? Can we choose a better variable name?

Reply:
Changed the name to mapProviders